src/libfetchers/git.cc

```diff
@@ -646,6 +649,11 @@ struct GitInputScheme : InputScheme
                 mounts.insert_or_assign(CanonPath::root, accessor);
                 accessor = makeMountedInputAccessor(std::move(mounts));
             }
+        } else {
+            /* If we don't have submodules and aren't doing export
+               ignore, then the tree hash is useful info to provide. */
```

Comment:
- How is it useful?
- `getSubmodulesAttr(input)` doesn't tell us anything about whether the repo has submodules. `false` doesn't imply anything, and `true` only correlates.